Dev

chinmaygireesh opened the pull request:
code structure changrd

chinmaygireesh committed:
refactor:changes the code structure

1.indroduces folder structure for classes
2.follows naming conventions in C#
3.indroduces virtual methods

chinmaygireesh committed:
Merge branch 'sachinskumar41:dev' into dev

sachinskumar41 commented:
Add proper PR message. Remove mentioning Dev

- [ ]

sachinskumar41 commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+bin
```
May not needed gitignore files to push

sachinskumar41 commented on Models/Administrative.cs:
```diff
@@ -0,0 +1,28 @@
+    {
+         
+        public string AdminNo    
+        {get;set;}
```
write get, set accessors on the same line of property for the auto implementation case

sachinskumar41 commented on Models/Supporting.cs:
```diff
@@ -0,0 +1,29 @@
+        public string Superior    
+        {get;set;}
+        public string Field
+        {get;set;}
```
same here. check previous comment

sachinskumar41 commented on Models/Teaching.cs:
```diff
@@ -0,0 +1,29 @@
+        public override void Display()
+        {   
+            base.Display();
+
```
remove unwanted spacing

sachinskumar41 commented on Program.cs:
```diff
@@ -0,0 +1,38 @@
+            {
+                case 1:
+                    Teaching techObj = new Teaching();
+                    techObj.Register();
```
There should be a submenu to choose the actions. Read the requirement again

chinmaygireesh committed:
refator:indruduces new feauturs

-changes the code structure
-indroduces action menu for staff users
  which contains
  1.add staff
  2.shoaw a staff
  3.show all staff
  4.Update a staff
  5.delete a staff

sachinskumar41 commented on Program.cs:
```diff
@@ -0,0 +1,45 @@
+    {
+        static void Main(string[] args)
+        {   
+            List<dynamic> staffList = new List<dynamic>();
```
Do we need a List in this case?

sachinskumar41 commented on Program.cs:
```diff
@@ -0,0 +1,45 @@
+            List<dynamic> staffList = new List<dynamic>();
+            int empId = 1;
+            while (true)
+            {   
```
This code is considered as a bad form, as it states that loops forever

sachinskumar41 replied:
When will it break out of the loop?

sachinskumar41 commented on Program.cs:
```diff
@@ -0,0 +1,45 @@
+                int userChoice = Convert.ToInt32(Console.ReadLine());
+                switch(userChoice) 
+                {
+                case 1:
```
Isn't our requirement states that the main menu is to manage staff followed by the operations?

sachinskumar41 replied:
Seems like it is reversed here

sachinskumar41 commented on Controller/MenuActions.cs:
```diff
@@ -0,0 +1,99 @@
+
+        public void DisplayAllStaffs(List<dynamic> staffList)
+        {   
+            foreach (dynamic staff in staffList)  
```
try to use the class property instead of dynamic

chinmaygireesh committed:
refactor:removes multiple staff reg. option

this commit will change
-main menu followed by action menu.
-uses class property instead of dynamic property
